catroid/src/main/java/org/catrobat/catroid/formulaeditor/FormulaElement.java

```diff
@@ -315,62 +317,44 @@ private Object interpretUserList(Sprite sprite) {
 		if (userListValues.size() == 0) {
 			return EMPTY_USER_LIST_INTERPRETATION_VALUE;
 		} else if (userListValues.size() == 1) {
-			Object userListValue = userListValues.get(0);
-			if (userListValue instanceof String) {
-				return userListValue;
-			} else {
-				return userListValue;
-			}
+			return userListValues.get(0);
 		} else {
 			return interpretMultipleItemsUserList(userListValues);
 		}
 	}
 
 	private Object interpretMultipleItemsUserList(List<Object> userListValues) {
 		List<String> userListStringValues = new ArrayList<>();
-		boolean concatenateWithoutWhitespace = true;
 
 		for (Object listValue : userListValues) {
 			if (listValue instanceof Double) {
 				Double doubleValueOfListItem = (Double) listValue;
-				if (isNumberAIntegerBetweenZeroAndNine(doubleValueOfListItem)) {
-					userListStringValues.add(doubleValueOfListItem.intValue() + "");
-				} else {
-					concatenateWithoutWhitespace = false;
-					userListStringValues.add(listValue.toString());
-				}
+				userListStringValues.add(stringWithoutTrailingZero(String.valueOf(doubleValueOfListItem.intValue())));
 			} else if (listValue instanceof String) {
 				String stringValueOfListItem = (String) listValue;
-				if (stringValueOfListItem.length() == 1) {
-					userListStringValues.add(stringValueOfListItem);
-				} else {
-					userListStringValues.add(stringValueOfListItem);
-					concatenateWithoutWhitespace = false;
-				}
+				userListStringValues.add(stringWithoutTrailingZero(stringValueOfListItem));
 			}
 		}
 
 		StringBuilder stringBuilder = new StringBuilder(userListStringValues.size());
-		boolean isFirstListItem = true;
+		String separator = listConsistsOfSingleCharacters(userListStringValues);
 		for (String userListStringValue : userListStringValues) {
```

Comment:
clean code wise it would be nicer to have that method return true or false and then go there with

String separator = listConsistsOfSingleCharacters(userListStringValues) ? "" : " ";

because method name implies to me that it would return true or false, not a separator.

Reply:
ok, done